packages/jupyterlab-gridstack/src/editor/gridstack/widget.ts

```diff
@@ -436,24 +479,38 @@ export class GridStackWidget extends Widget {
         return;
       }
 
-      const row = Math.floor(event.offsetY / this.layout.cellHeight);
+      const row = Math.floor(
+        (event.offsetY + this.node.scrollTop) / this.layout.cellHeight
+      );
       const col = Math.floor(
         (this.layout.columns * event.offsetX) / this.node.offsetWidth
       );
-      const width = this.layout.columns - col;
+      let width = 1;
       let height = 1;
       if (widget!.model.type === 'code') {
-        height = Math.ceil(
-          (widget!.node.scrollHeight - widget!.inputArea.node.scrollHeight) /
-            this.layout.cellHeight
+        // The constant 40, is the ~size of the toolbar in px
+        const rect = (
+          widget as CodeCell
+        ).outputArea.node.getBoundingClientRect();
+        const c = this.layout.columns - col;
+        width = Math.min(
+          c,
+          Math.ceil(rect.width / this.layout.grid.cellWidth())
         );
+        height = Math.ceil((rect.height + 40) / this.layout.cellHeight);
       } else {
-        height = Math.ceil(widget!.node.scrollHeight / this.layout.cellHeight);
+        const rect = widget!.node.getBoundingClientRect();
+        const c = this.layout.columns - col;
+        width = Math.min(
+          c,
+          Math.ceil(rect.width / this.layout.grid.cellWidth())
+        );
+        height = Math.ceil((rect.height + 40) / this.layout.cellHeight);
```

Comment:
Wondering whether the `40` could be hoisted in a variable, or we add a small comment?

It looks like the same number is used in other places so it could be useful to explain why we choose that value.

Reply:
There is a comment, it is the size of the toolbar.

Reply:
ouch, sorry. I forgot to add the comment in this function. I'll add

Reply:
Ah right I missed it.

Reply:
I added a constant with the size, probably better than the comment.

Reply:
:+1: